### DISCUSSION TITLE
fly

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#521] [Q&A] [answered]
i want write function to fly the view:
1.open the model on the web
2.move or change angle or zoom by the web tools
3.save the DATA( i think include current camera info  and so on  ) to list
4.loop 2 and 3 step
5.load  and play the different angle from the DATA 
so,i donot know what and  how save the DATA and how load the DATA

**Comment by pmconne** [comment, marked as answer]
[ViewState.toProps](https://github.com/imodeljs/imodeljs/blob/a14573698473cbe759f8a8da705a11ed54494486/core/frontend/src/ViewState.ts#L196) will give you a JSON representation of the view that you can save for later reuse.
To restore the view and apply it to a Viewport, do something like:
```
const viewClass = await imodel.findClassFor<typeof ViewState>(savedViewProps.viewDefinitionProps.classFullName, undefined);
if (viewClass) {
  const view = viewClass.createFromProps(savedViewProps, imodel);
  await view.load();
  viewport.changeView(view);
}
```
